compiler/nim.nim

```diff
@@ -74,22 +74,23 @@ proc handleCmdLine(cache: IdentCache; conf: ConfigRef): CmdLineHandlingResult =
     processCmdLine: processCmdLine
   )
   self.initDefinesProg(conf, "nim_compiler")
-  if paramCount() == 0:
+  if argv.len == 0:
     return cliErrNoParamsProvided
 
-  self.processCmdLineAndProjectPath(conf)
+  self.processCmdLineAndProjectPath(conf, argv)
   if conf.errorCounter != 0: return
+  
   let graph = newModuleGraph(cache, conf)
-
-  if not self.loadConfigsAndProcessCmdLine(cache, conf, graph) or
-      conf.errorCounter != 0:
+  
+  if not self.loadConfigsAndProcessCmdLine(cache, conf, graph, argv):
     return
+  if conf.errorCounter != 0: return
```

Comment:
Please also merge this line back into the above `if`.

Reply:
@iacore please restore this

Reply:
The semantic is the same. Should we make the code clearer?

Reply:
It is and making the code clearer is good; I'm not sure how splitting them apart does that, maybe you meant another way entirely?

The idea behind restoring the line stems from small PR hygiene and keeping file history clean for those spelunking through trying to debug/understand things (false positive).

Reply:
Following up in chat to discuss this, I don't think we need to block the PR for this.